### PR TITLE
frontend performance queue updates 500ms interval

### PR DESCRIPTION
**Pull request by jamestalton**
Signed-off-by: James Talton <jtalton@redhat.com>

### CODE REVIEW

File: frontend/src/atoms.tsx [hunk 1]
```diff
@@ -192,63 +192,46 @@ export function LoadData(props: { children?: ReactNode }) {
...
-        async function processEvents() {
-            if (!eventDataQueue) return
-            const dataToProcess = eventDataQueue
-            for (const kind in setters) {
+        function processEventQueue() {
+            if (eventQueue.length === 0) return
+
+            const kindsMap = eventQueue?.reduce((kindsMap, eventData) => {
+                const kind = eventData.object.kind
+                if (kind) {
```
Comment (KevinFCormier) [inline frontend/src/atoms.tsx:202]: Is kind ever falsy?

Comment (KevinFCormier) [reply frontend/src/atoms.tsx:202]: Also, I guess we don't currently support duplicate kinds in different apiGroups?

File: frontend/src/atoms.tsx [hunk 2]
```diff
@@ -293,7 +275,10 @@ export function LoadData(props: { children?: ReactNode }) {
             }
         }
         startWatch()
+
+        const timeout = setInterval(processEventQueue, 1000)
```
Comment (KevinFCormier) [inline frontend/src/atoms.tsx:279]: Have you played with different values? Would half a second work? And maybe this should be stored in configuration?

**Review by KevinFCormier — COMMENTED** [review]
A couple comments.

File: frontend/src/atoms.tsx (hunk 2 above)
Comment (jamestalton) [reply frontend/src/atoms.tsx:279]: A half sec would work.

File: frontend/src/atoms.tsx (hunk 1 above)
Comment (jamestalton) [reply frontend/src/atoms.tsx:202]: kind should never be falsy, but sometimes I code defensively.
don't currently support duplicate kinds

**Pull request merged**